feature/share-task-code

free2fqn opened the pull request:
Я все еще в процессе, но кое-что сделал.
Столкнулся с двумя проблемами: 

1. Почему-то у меня все еще открывается морда 13.html (как и любая другая) на "http://localhost:63342/platform/...", а она не может стучаться на наши контроллеры по адресам вида "/example-endpoint". Раньше я настраивал CORS и обозначал жестко адрес сервиса, на котором эндпоинт, но @nekromant322 бил по рукам за это. Я не смог это побороть.

2. У меня не получается заинжектить код ученика в редактор.

free2fqn committed:
added 'help-button' in 13.html

free2fqn committed:
Java 8 -> Java 11 migration (README and build.yml update)

free2fqn committed:
Added Coffeine Cache support in Orchestrator's pom.xml.
Created OrchestratorBeanConfig (needs to config Cache bean).
Created HelpMeTaskTextAndCodeDTO.
Created cache.timeout field in Orchestrator's application.yml.
Moved "Sample Input/Output" into <div> in 13.html.
Added code.js in 13.html (didn't work previously)
Added function getTaskTextAndCodeDTO() for "help button" in code.js.
Created HelpMeRestController.

free2fqn committed:
Added bean-field helpMeCache, added endpoint "/helpMe/{key}" in PageController.
Created helpMeView.html and emptyHelpMeView.html.
Endpoint "/helpMe" in HelpMeRestController now returns int hash-key for Cache.
Clean up code.js.
Clean up 13.html.

nekromant322 commented on platform_dto/src/main/java/dtos/HelpMeTaskTextAndCodeDTO.java:
```diff
@@ -0,0 +1,9 @@
+
+@Data
+public class HelpMeTaskTextAndCodeDTO {
+    private String taskText;
```
taskHtml?

nekromant322 commented on platform_orchestrator/src/main/java/com/override/controller/PageController.java:
```diff
@@ -21,4 +33,14 @@ public String balancePage() {
+        if (helpMeDTO == null) {
+            return "emptyHelpMeView";
+        }
+        model.addAttribute("insertText", helpMeDTO.getTaskText());
```
Алло, никакого themlyeaf

nekromant322 commented on platform_orchestrator/src/main/java/com/override/util/OrchestratorBeanConfig.java:
```diff
@@ -0,0 +1,27 @@
+    Cache<Integer, HelpMeTaskTextAndCodeDTO> getCacheBean() {
+        Cache<Integer, HelpMeTaskTextAndCodeDTO> resultCache = Caffeine.newBuilder()
+                .expireAfterWrite(timeToExpireCache, TimeUnit.MINUTES)
+                .maximumSize(100)
```
Вынести в настройки?

nekromant322 commented:
А ты попробуй http://localhost:8000/lessons/core/1/2/13

free2fqn committed:
Added cache.size field in application.yml and injected it into OrchestratorBeanConfig.
Renamed taskTEXT -> taskHTML in HelpMeTaskTextAndCodeDTO.

free2fqn committed:
Added copying to clipboard link to helpMeView.html (message about it writes in console.log).
Clean up

free2fqn committed:
Clean up code.js, PageController.
Added endpoint "/helpMe/cache/{key}". It will send HelpMeTaskTextAndCodeDTO, if it's key is now cached.
 Removed thymeleaf from helpMeView.html.
 Created helpMeView.js.

free2fqn committed:
added modal window with field for student's question in 13.html.
Refactored code.js for modal window.
Clean ip HelpMeRestController from console output - using Slf4j now.
Added field question in renamed HelpMeTaskTextAndCodeDTO -> HelpMeDTO.
Added question-field in helpMeView.html

free2fqn committed:
created and added fragment helpMeButton.html in 13.html
